### ISSUE TITLE
Blazor Hot Reload only Works with dotnet watch, not from Visual Studio 2022

### ISSUE THREAD

**Issue by MichaelPeter** [dotnet/aspnetcore#38202]
### Describe the bug

Blazor Hot Reload only works with dotnet watch, not from Visual Studio 2022.

### To Reproduce
- Create a new Blazor Webassembly App Project with .NET 6 (Mark as ASP.NET Core hosted.)
- Start the Projekt
- Change the Hello World to Hello World 2 in the Index.razor and save.
- Press the hot reload button

The checkmark appears in the blazor UI but  **the Text is not updated.** Also Visual Studio writes in the toolbar changes applied.

Changes made in the aspnetcore server project are applied

I also tried to specifiy the hotReloadProfile blazorwasm  in launch.json for the Client Project and aspnetcore for the Server Project but still blazor was not updated.

Do I forget something?

but if I goto the Server project folder and execute **dotnet watch** and change anything
the **changes get applied**. hotReloadProfile in launch.json is not required.

- ASP.NET Core version: 6.0.0
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version: Visual Studio 2022 17.1 Preview 1.0 (Preview with Release)
- Dotnet Version 6.0
- Dotnet Info: Not right now on the pc.

**Comment by pranavkm** [comment]
Thanks for the issue report @MichaelPeter. Could you use the [VS feedback tool](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022) to report the issue? That way it will get to the routed to the right that owns this experience in VS.

**Comment by khoughton-Landmark** [comment]
I also have this exact same experience.

**Comment by MichaelPeter** [comment]
Ok Microsoft, your communication here is just terrible:

Here is a video posted 4 hours ago: (notice blazor is not in the debugger)
https://www.youtube.com/watch?v=3LJDMAOm8JU&t=96s

Here is a very cryptically written blogpost
https://devblogs.microsoft.com/dotnet/update-on-net-hot-reload-progress-and-visual-studio-2022-highlights/

What is happening:
Hot Reload for Blazor wasm doesn't work wihtout the debugger - either click start without debugger - or run dotnet watch
But none of these article says that directly....

**Comment by khoughton-Landmark** [comment]
You, sir, are a life-saver.  I thought I was going crazy. I kept seeing the magic hot reload working in videos and for the life of me could not figure out WHY it wasn't working for me.  I ALWAYS run in debug mode when developing. Perhaps that's a bad habit, but that's what I do.  I just tested hot reload WITHOUT debugger and it works as expected.

Microsoft, the next thing will be to combine Hot Reload AND Debugging....  I mean, really.... wouldn't you want to set a breakpoint on a possible bug, and fix it in REAL TIME?

> Ok Microsoft, your communication here is just terrible:
> 
> Here is a video posted 4 hours ago: (notice blazor is not in the debugger) https://www.youtube.com/watch?v=3LJDMAOm8JU&t=96s
> 
> Here is a very cryptically written blogpost https://devblogs.microsoft.com/dotnet/update-on-net-hot-reload-progress-and-visual-studio-2022-highlights/
> 
> What is happening: Hot Reload for Blazor wasm doesn't work wihtout the debugger - either click start without debugger - or run dotnet watch But none of these article says that directly....

**Comment by cdc1671** [comment]
Just my 2 cents... Have 2 dev machines, my main laptop Windows 10 and my home desktop which I just did a clean install of windows 11 ... VS2022+Hot Reload is working fine on the clean install but fails on the Win10 machine.

**Comment by MichaelPeter** [comment]
> Just my 2 cents... Have 2 dev machines, my main laptop Windows 10 and my home desktop which I just did a clean install of windows 11 ... VS2022+Hot Reload is working fine on the clean install but fails on the Win10 machine.

But hot reload in debug mode also doesn't work for you right?

**Comment by BenWhite27** [comment]
Just adding my somewhat related experience here as this comes up in searches a fair bit.
My existing Blazor WebAssembly project which I migrated to .NET 6 wasn't working through Ctrl+F5, the hot reload output in VS2022 reported that the Server hot reload session had started but not the WebAssembly one.  Took a while but finally found that somewhere in the project reference chain there was a project that still targeted .NET 5.  I was able to remove this reference and hot reload started working me.  I may have cleaned the solution and restarted VS2022 as well.

@MichaelPeter correct hot reload for blazor wasm isn't supported right now: https://docs.microsoft.com/en-us/aspnet/core/test/hot-reload?view=aspnetcore-6.0#blazor-webassembly you're right this could use some more emphasis on the page.

> In Visual Studio 2022 GA (17.0), Hot Reload is only supported when running without the debugger.

**Comment by wgrs** [comment]
Why does microsoft deliberately mislead people into thing that Blazor WASM support hot reload? It does not and never did. Why not just publish a doc stating this clearly instead of pretending it does just to keep the corp bosses happy. You are wasting everones time!

**Comment by kevonh** [comment]
> Why does microsoft deliberately mislead people into thinking that Blazor WASM supports hot reload? It does not and never did. Why not just publish a doc stating this clearly instead of pretending it does just to keep the corp bosses happy. You are wasting everones time!

It actually does work... just not in debug mode, and not if you have response compression turned on.  There are a few other caveats, but once you have your environment in the right mode, it works pretty well.  

What would be NICE is a document detailing all of these caveats that went along with the announcements, so it's CLEAR to everyone what you have to do in order to get it to work.

**Comment by wgrs** [comment]
> > Why does microsoft deliberately mislead people into thinking that Blazor WASM supports hot reload? It does not and never did. Why not just publish a doc stating this clearly instead of pretending it does just to keep the corp bosses happy. You are wasting everones time!
> 
> It actually does work... just not in debug mode, and not if you have response compression turned on. There are a few other caveats, but once you have your environment in the right mode, it works pretty well.
> 
> What would be NICE is a document detailing all of these caveats that went along with the announcements, so it's CLEAR to everyone what you have to do in order to get it to work.

Its does not work. I have been testing extensively today and not once has the hot reload worked. I tested throughout preview and it also did not work correctly. I am testing with the latest of everything ( not preview ). Stack Overflow is full of failed attempts to get it to work. I have also tested the CLI commands - using the latest CLI .

**Comment by BenWhite27** [comment]
> Its does not work in any mode without restarting the app
I have Hot Reload working quite well for probably 80% of my development. Sure when I make substantial changes to the code I expect it to ask me to restart the app but for all those little tweaks I make to UI, CSS and method bodies it works great and has saved me a bunch of time.

If you're having difficulty, try creating a blank project and testing hot reload in there it's likely to be something in your project that's making it not work properly.  If so try creating a new issue here as this closed one will probably not get re-opened.

**Comment by wgrs** [comment]
I am using a brand new Blazor project - straight form the template. I have submitted so many bug reports I dont bother anymore - surely one look at Stack Overflow will tell them its not working.

**Comment by martonx** [comment]
I tried VS 2022 Preview 2.0 and Hot Reload and Debugging  in parallel, but it doesn't work yet. Very annoying when you change some little thing in your razor C# code, while you are standing in a debug session, and you have to restart the full application.
